M/MAGMA/build_tarballs.jl

```diff
@@ -1,56 +1,44 @@
-using BinaryBuilder
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
 
-# Collection of sources required to build Nettle
 name = "MAGMA"
-version = v"2.5.1"
+version = v"2.5.2"
+
+# Collection of sources required to complete build
 sources = [
-    "http://icl.utk.edu/projectsfiles/magma/downloads/magma-2.5.1.tar.gz" =>
-    "ce32c199131515336b30c92a907effe0c441ebc5c5bdb255e4b06b2508de109f",
-#    "./bundled",
+    ArchiveSource("https://github.com/Roger-luo/MAGMA/archive/v2.5.2.tar.gz", "dc3b90182825721d9b9ca1deb8afb2ff547ee41991ed698225832693a30a64ab")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/magma-*/
-
-# We need a newer version of FindCUDA.cmake
-curl -L 'https://raw.githubusercontent.com/Kitware/CMake/v3.13.0/Modules/FindCUDA.cmake' -o /usr/share/cmake/Modules/FindCUDA.cmake
-
-# Apply patches to force name-mangling
-atomic_patch -p1 ${WORKSPACE}/srcdir/patches/fortran_mangling.patch
-
-mkdir build && cd build
-export CFLAGS="${CFLAGS} -DMAGMA_ILP64"
-cmake .. -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
-    -DLAPACK_LIBRARIES="-lopenblas64_ -lgfortran -lquadmath" \
-    -DCUDA_TOOLKIT_ROOT_DIR="${prefix}" \
-    -DCUDA_TOOLKIT_TARGET_DIR="${prefix}" \
-    -DGPU_TARGET="Maxwell Pascal Volta" \
-    -DCUDA_TOOLKIT_ROOT_DIR_INTERNAL="${prefix}"
-
-make -j${nproc}
-make install
+export CUDADIR="${prefix}/cuda"
+export OPENBLASDIR="${prefix}"
+export PATH=${PATH}:${WORKSPACE}/destdir/cuda/bin
+cd ${WORKSPACE}/srcdir/MAGMA-2.5.2/
+make lib -j${nproc}
+make install -j${nproc}
+exit
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = [
-    Linux(:x86_64),
-    Windows(:x86_64),
-    MacOS(:x86_64),
+    Linux(:x86_64, libc=:glibc)
 ]
-platforms = expand_gcc_versions(platforms)
+
 
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("libMAGMA", :libMAGMA),
+    LibraryProduct("libmagma.so", :libmagma)
```

Comment:
```suggestion
    LibraryProduct("libmagma", :libmagma)
```
The problem is not the filename, which was correct without extension, but the fact that the library can't be dlopened:
```
[ Info: Checking shared library lib/libmagma.so
ERROR: could not load library "/agent/_work/1/s/M/MAGMA/build/x86_64-linux-gnu/QUIlwQWa/destdir/lib/libmagma.so"
libcublasLt.so.10: cannot open shared object file: No such file or directory
[ Info: Set SONAME of lib/libmagma.so to "libmagma.so"
[ Info: Checking shared library lib/libmagma_sparse.so
ERROR: could not load library "/agent/_work/1/s/M/MAGMA/build/x86_64-linux-gnu/QUIlwQWa/destdir/lib/libmagma_sparse.so"
libcublasLt.so.10: cannot open shared object file: No such file or directory
```